### PR TITLE
feat: abstract dataTable into reusable component

### PR DESCRIPTION
**Pull request by epicallan**

### CODE REVIEW

File: src/views/components/DataTable/index.tsx [hunk 1]
```diff
@@ -219,13 +70,13 @@ export default class DataTable extends React.Component<Props, State> {
     this.csvLink = React.createRef();
```
Comment (aleku399) [inline src/views/components/DataTable/index.tsx]: this.reactTable is not defined

File: src/views/components/DataTable/index.tsx [hunk 2]
```diff
@@ -5,114 +5,40 @@ import ReactTable, { Column, Filter } from "react-table";
...
-  data: EvaluationTableData[];
+export interface State<T> {
+  data: Array<T & TableData>;
   filtered: Filter[];
   search: string;
   page: number;
   pageSize: number;
   from: string;
   to: string;
+  dataToDownload: any[];
```
Comment (aleku399) [inline src/views/components/DataTable/index.tsx]: why is there no type for dataToDownload

Comment (epicallan) [reply src/views/components/DataTable/index.tsx]: I haved added it

File: src/views/components/DataTable/index.tsx (hunk 1 above)
Comment (epicallan) [reply src/views/components/DataTable/index.tsx]: It is defined. It's above `this.csvLink`

**Pull request merged**